CMakeLists.txt

```diff
@@ -1,303 +1,159 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
-PROJECT(ARRAYFIRE)
+cmake_minimum_required(VERSION 3.5.2)
+project(ArrayFire
+  VERSION 3.5.0
+  LANGUAGES C CXX )
```

Comment:
We should be doing it the other way around. This is the main location where the version should be set. CMake will make a variable named ArrayFire_VERSION_MAJOR based off of this value. I have changed the versioning source to reflect this.